### PR TITLE
Step 1/3. add dummy semantic segmentation training script.

### PR DESCRIPTION
**Pull request by tanzhenyu**
Step 1: add training script that works -- without 2 & 3, this gives meanIOU ~ 22%
Step 2: add SBD data
Step 3: add random resize data augmentation that improves training

### CODE REVIEW

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        y_true = y_true.to_tensor()
+    return image, y_true
+
+
+train_ds = train_ds.map(pad_fn, num_parallel_calls=tf.data.AUTOTUNE)
+eval_ds = eval_ds.map(pad_fn, num_parallel_calls=tf.data.AUTOTUNE)
+
+with strategy.scope():
+    backbone = keras_cv.models.ResNet50V2(
+        include_rescaling=True, weights="imagenet/classification-v2", include_top=False
```
Comment (ianstenbit) [inline examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py:75]: (optional): use just `"imagenet"` to use the best ImageNet weights (in its current form it will pin to v2)

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    ).as_backbone()
+    model = DeepLabV3(classes=21, backbone=backbone, include_rescaling=True)
```
Comment (ianstenbit) [inline examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py:77]: Question about rescaling:

if the backbone is rescaling and the DeepLab model is also rescaling, are we double-rescaling?

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    optimizer = tf.keras.optimizers.SGD(
+        learning_rate=0.0001, momentum=0.9, clipnorm=2.0
```
Comment (ianstenbit) [inline examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py:79]: Should this be flag-controlled? (Maybe also batch size?)
(Fine with me if this will come in part 2 or 3)

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(
+        from_logits=True, reduction=tf.keras.losses.Reduction.NONE
+    )
+
+    metrics = [
+        tf.keras.metrics.SparseCategoricalCrossentropy(),
+        tf.keras.metrics.MeanIoU(num_classes=21, sparse_y_pred=False),
+    ]
+model.compile(optimizer, loss="SparseCategoricalCrossentropy", metrics=metrics)
+model.fit(train_ds, epochs=50, validation_data=eval_ds)
```
Comment (ianstenbit) [inline examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py:90]: Should we add callbacks for BackupAndRestore and ModelCheckpoint?

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
+# Copyright 2022 The KerasCV Authors
```
Comment (ianstenbit) [inline examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py:1]: Can we move this to `examples/training/semantic_segmentation/pascal_voc/deeplab_v3.py`?

(Or something generic in that folder like `basic_training.py` if this can support other models)

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks!

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py (hunk 5 above)
Comment (tanzhenyu) [reply examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py:1]: Sounds good. task/dataset/model is better than task/model/dataset

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py (hunk 2 above)
Comment (tanzhenyu) [reply examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py:77]: IMHO we should not expose `include_rescaling` at meta arch level, if it accepts `backbone` as another argument. However the trick here is when `backbone` also accept a string we don't know whether we should include that or not. Not particularly related to this PR, but we should definitely discuss this and have a consensus.

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py (hunk 3 above)
Comment (tanzhenyu) [reply examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py]: Done.

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py (hunk 4 above)
Comment (tanzhenyu) [reply examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py]: Added ModelCheckpoint.

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py (hunk 1 above)
Comment (tanzhenyu) [reply examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py]: Done. I assume if the weights is not found, error will be thrown?

**Review by qlzh727 — APPROVED** [review]
The metric calculation might need some update, to ignore the class 255 (boundary).

**Comment by tanzhenyu** [comment]
> The metric calculation might need some update, to ignore the class 255 (boundary).

It is currently ignored in the class_segmentation_map during data pipeline as background mask. But yeah good point! We should move this to metrics part instead of data pipeline.

File: examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py (hunk 2 above)
Comment (ianstenbit) [reply examples/models/semantic_segmentation/deeplab_v3/pascal_voc/train.py]: I agree that this should be backbone-configured (maybe that means for the meta-arch we have an optional `include_rescaling` which should be supplied iff using a string backbone)

**Review by ianstenbit — APPROVED** [review]
LGTM
